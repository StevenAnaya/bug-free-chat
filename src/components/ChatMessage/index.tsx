import styled from 'styled-components';

interface Props {
	isMine?: boolean;
}

export const StyledChatMessage = styled.div<Props>`
	/* DEFAULT DEFINITIONS */
	display: flex;
	align-items: center;
	width: fit-content;
	background-color: ${({ theme, isMine }) => isMine ? theme.color.lightPurple : theme.color.gray};
	min-height: 30px;
	padding: 2px 10px;
	border-radius: ${({ theme }) => theme.misc.bRadius};
	font-family: ${({ theme }) => theme.fonts.family};
	font-size: ${({ theme }) => theme.fonts.size.sm};
	color: ${({ theme, isMine }) => isMine ? theme.color.softGray : theme.color.darkGray};
`;
