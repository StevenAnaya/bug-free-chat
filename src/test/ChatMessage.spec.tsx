import React from 'react';
import { render } from 'test/utils/customRender';

import { StyledChatMessage } from 'components/ChatMessage';

import 'jest-styled-components';

describe('[ChatMessage]', () => {
	test('Render a no viewed ChatMessage Component', () => {
		const { container } = render(<StyledChatMessage>Some Text</StyledChatMessage>);

		expect(container.firstChild).toMatchSnapshot();
	})

	test('Render a viewed ChatMessage Component', () => {
		const { container } = render(<StyledChatMessage viewed>Some Text</StyledChatMessage>);

		expect(container.firstChild).toMatchSnapshot();
	})
});
