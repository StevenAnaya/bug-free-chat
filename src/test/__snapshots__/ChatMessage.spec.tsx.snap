// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ChatMessage] Render a no viewed ChatMessage Component 1`] = `
.c0 {
  background-color: #35234e;
  min-height: 60px;
}

<div
  class="c0"
>
  Some Text
</div>
`;

exports[`[ChatMessage] Render a viewed ChatMessage Component 1`] = `
.c0 {
  background-color: #8255b6;
  min-height: 60px;
}

<div
  class="c0"
>
  Some Text
</div>
`;
